backend/instancemgmt/instance_manager.go

```diff
@@ -119,7 +122,13 @@ func (im *instanceManager) Get(ctx context.Context, pluginContext backend.Plugin
 		}
 
 		if disposer, valid := ci.instance.(InstanceDisposer); valid {
-			disposer.Dispose()
+			time.AfterFunc(disposeTTL, func() {
+				log.DefaultLogger.Debug("Disposing instance", "cacheKey", cacheKey)
+				disposer.Dispose()
+				activeInstances.Dec()
+				log.DefaultLogger.Debug("Instance disposed", "cacheKey", cacheKey)
```

Comment:
Hmm... Not sure about leaking the cache key. Probably mostly paranoia, but would omit unless really necessary

Reply:
Hmm yeah I was afraid you was going to comment on that :) But for debug purposes the reasoning behind adding logs, we need to be able to identity the instances. If we cannot use the cache key I'll probably skip adding any log messages since we would need to extract things from plugincontext otherwise

Reply:
Considering it's at debug level maybe it's fine then, since if someone wanted to find this maliciously, they'd need quite a bit of control

Reply:
I removed it for now